src/adagrams.js

```diff
@@ -1,15 +1,144 @@
 export const drawLetters = () => {
   // Implement this method for wave 1
+  let lp_copy = JSON.parse(JSON.stringify(LETTER_POOL));
+  let drawLetters = [];
+  const sum = sumOfValues(lp_copy);
+  while (drawLetters.length < 10) {
+    const index = Math.floor(Math.random() * (sum + 1));
+    const letter = whichLetter(index, lp_copy);
+    //const letter = Object.keys(lp_copy)[index];
+    if (lp_copy[letter] > 0) {
+      drawLetters.push(letter);
+      lp_copy[letter]--;
+    }
+  }
+  return drawLetters;
+};
+
+const whichLetter = (index, lp_copy) => {
+  for (let letter of Object.keys(lp_copy)) {
+    if (index === 0) {
```

Comment:
I would like to understand this code better. I've taken about 10 minutes to try to puzzle out the purpose of `index` and why it decrements.  I'm guessing that you sat with this code for some time! Maybe this could be a good example of how judicious uses of comments can help clarify code :)

Reply:
I should have added more comments for sure. 

 index is the random value that is generated between 0 and 98 (98 is the sum of all the values in the letter_pool- it is calculated in a separate helper function.). So a letter with more values in the letter_pool has a higher probability of draw. This helps so we can get more vowels to make words with.

The whichLetter function, finds the letter associated with the index in the letter_pool. It keeps on decrementing the index by the value of the letters found in the letter_pool. We keep searching till index == 0, that's when the current letter is the one that's at the randomly picked index and we return it.